Docker plans?

Amazing work here, thank you!

Are there any plans or would there be any consideration made towards compiling your code into a Docker container to streamline set up?

I had no difficulty building and launching the container, but I'm finding that the launch files don't appear to find my dataset files. My dataset files are placed in the following structure:
```
|-----rt_of_low_high_res_event_cameras
|--launch
|--test_vids
|  |--prophesee
|  |  |--checkerboard.bag
|--src
....
```
When I run 
`roslaunch rt_of_low_high_res_event_cameras prophesee_gen4_replay.launch
`I run into...
`[ERROR] [1734025957.767291839]: Unknown input format dat/file, exiting!`

Where my launch XML looks like 
```
<launch>
  <!-- Edges node -->
  <node name="edges" pkg="rt_of_low_high_res_event_cameras" type="edges_node_replay" ns="rt_of_low_high_res_event_cameras" output="screen">
    <param name="input_file" type="string" value="$(find rt_of_low_high_res_event_cameras)/test_vids/prophesee/checkerboard.bag"/>
    <param name="topic" type="string" value="/prophesee/camera/cd_events_buffer"/>
    <param name="accumulation_window" type="int" value="15"/>
    <param name="height" type="int" value="720"/>
    <param name="width" type="int" value="1280"/>
    <param name="queues_size" type="int" value="10000"/>
    <!--<param name="calibration_file" type="str" value="..."/>-->
  </node>

  <!-- Denoising & filling node -->
  <node name="denoising_filling" pkg="rt_of_low_high_res_event_cameras" type="denoising_filling_node" ns="rt_of_low_high_res_event_cameras" output="screen">
    <param name="use_gpu_version" type="bool" value="true"/>
    <param name="denoising_min_neighbours" type="int" value="2"/>
    <param name="filling_min_neighbours" type="int" value="3"/>
    <param name="queues_size" type="int" value="10000"/>
  </node>

  <!-- Distance surface node -->
  <node name="distance_surface" pkg="rt_of_low_high_res_event_cameras" type="distance_surface_node" ns="rt_of_low_high_res_event_cameras" output="screen">
    <param name="use_gpu_version" type="bool" value="true"/>
    <param name="distance_surface_formulation" type="string" value="exponential"/>
    <param name="queues_size" type="int" value="10000"/>
  </node>

  <!-- Optical flow node -->
  <node name="optical_flow" pkg="rt_of_low_high_res_event_cameras" type="optical_flow_node" ns="rt_of_low_high_res_event_cameras" output="screen">
    <param name="pyramidal_flow_levels" type="int" value="3"/>
    <param name="max_flow" type="double" value="15.0"/>
    <rosparam param="gammas">[500.0, 500.0, 500.0]</rosparam>
    <rosparam param="smooth_iterations">[20, 20, 20]</rosparam>
    <param name="queues_size" type="int" value="10000"/>
  </node>

  <!-- Optical flow visualization node -->
  <node name="optical_flow_viz" pkg="rt_of_low_high_res_event_cameras" type="optical_flow_viz_node" ns="rt_of_low_high_res_event_cameras" output="screen">
    <param name="visualization_method" type="string" value="colors"/>
    <param name="clip_flow" type="double" value="-1"/>
    <param name="queues_size" type="int" value="10000"/>
  </node>

  <!-- Display -->
  <node name="image_view" pkg="rqt_image_view" type="rqt_image_view" args="/rt_of_low_high_res_event_cameras/optical_flow_viz"/>

</launch>
```
All I've modified here is the **input_file** parameter under the **edge** node. I was unsure whether this was a ROS versioning issue? Thank you for your time.

Best,
Stephen

Thanks for the quick response!

It turns out that this specific error was indeed because I altered the file path after loading by mistake--thanks for suggesting that. However, rebuilding the container it seems my main issue is not yet resolved (though the `[ERROR] [1734025957.767291839]: Unknown input format dat/file, exiting!` statement is now absent):

`could not connect to display 
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.`

Presumably the file path is correct now, but there still appears to be a package issue. Any guesses on what might be happening here? Thanks again for your time and help. 

Best,
Stephen

--- a/Dockerfile
+++ b/Dockerfile
@@ -6,11 +6,11 @@ SHELL ["/bin/bash", "-c"]
 
 # We install some required packages
 RUN apt update
-RUN DEBIAN_FRONTEND=noninteractive TZ=Etc/UTC apt install -y build-essential curl git nano software-properties-common tzdata
+RUN DEBIAN_FRONTEND=noninteractive TZ=Etc/UTC apt install -y build-essential git nano software-properties-common tzdata wget
 
 # We setup ROS by following the wiki instructions (https://wiki.ros.org/noetic/Installation/Ubuntu)
 RUN echo "deb http://packages.ros.org/ros/ubuntu focal main" > /etc/apt/sources.list.d/ros-latest.list
-RUN curl -s https://raw.githubusercontent.com/ros/rosdistro/master/ros.asc | apt-key add -
+RUN wget --no-check-certificate -qO - https://raw.githubusercontent.com/ros/rosdistro/master/ros.asc | apt-key add -
 RUN apt update
 RUN apt install -y ros-noetic-ros-base ros-noetic-catkin ros-noetic-rqt-image-view python3-catkin-tools
 RUN echo "source /opt/ros/noetic/setup.bash" >> ~/.bashrc

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,0 +1,50 @@
+# We use a Ubuntu 20.04 / CUDA 12.6 image as a base
+FROM nvidia/cuda:12.6.2-devel-ubuntu20.04
+
+# We change the default shell from sh to bash
+SHELL ["/bin/bash", "-c"]
+
+# We install some required packages
+RUN apt update
+RUN DEBIAN_FRONTEND=noninteractive TZ=Etc/UTC apt install -y build-essential curl git nano software-properties-common tzdata
+
+# We setup ROS by following the wiki instructions (https://wiki.ros.org/noetic/Installation/Ubuntu)
+RUN echo "deb http://packages.ros.org/ros/ubuntu focal main" > /etc/apt/sources.list.d/ros-latest.list
+RUN curl -s https://raw.githubusercontent.com/ros/rosdistro/master/ros.asc | apt-key add -
+RUN apt update
+RUN apt install -y ros-noetic-ros-base ros-noetic-catkin ros-noetic-rqt-image-view python3-catkin-tools
+RUN echo "source /opt/ros/noetic/setup.bash" >> ~/.bashrc
+RUN source ~/.bashrc
+
+# We setup the optical-flow-filter library (https://github.com/jadarve/optical-flow-filter)
+RUN git clone https://github.com/jadarve/optical-flow-filter.git
+WORKDIR /optical-flow-filter
+RUN mkdir build
+WORKDIR /optical-flow-filter/build
+RUN cmake ..
+RUN make install
+
+# We setup the catkin workspace (https://wiki.ros.org/catkin/Tutorials/create_a_workspace)
+WORKDIR /
+RUN mkdir -p catkin_ws/src
+WORKDIR /catkin_ws
+RUN catkin config --init --mkdirs --extend /opt/ros/noetic --merge-devel --cmake-args -DCMAKE_BUILD_TYPE=Release
+
+# We setup the rpg_dvs_ros package (https://github.com/uzh-rpg/rpg_dvs_ros)
+RUN add-apt-repository ppa:inivation-ppa/inivation
+RUN apt install -y libcaer-dev ros-noetic-cv-bridge ros-noetic-image-common
+WORKDIR /catkin_ws/src
+RUN git clone https://github.com/catkin/catkin_simple.git
+RUN git clone https://github.com/uzh-rpg/rpg_dvs_ros.git
+
+# We setup our rt_of_low_high_res_event_cameras package (https://github.com/heudiasyc/rt_of_low_high_res_event_cameras)
+RUN mkdir /catkin_ws/src/rt_of_low_high_res_event_cameras
+WORKDIR /catkin_ws/src/rt_of_low_high_res_event_cameras
+COPY ./ /catkin_ws/src/rt_of_low_high_res_event_cameras/
+
+# We build the rt_of_low_high_res_event_cameras package
+WORKDIR /catkin_ws
+RUN catkin build rt_of_low_high_res_event_cameras
+
+# Finally, when running the container, we should source the catkin workspace
+RUN echo "source devel/setup.bash" >> ~/.bashrc

--- a/README.md
+++ b/README.md
@@ -24,7 +24,30 @@ Our dataset can be downloaded from the following webpage: <https://datasets.hds.
 
 In this work, we propose an optimized framework for computing optical flow in real-time, with both low- and high-resolution event cameras, by computing dense image-like representations from the events through the use of our "inverse exponential distance surface".
 
-## Installing and compiling the code
+## Installing and compiling the code (with Docker)
+
+The easiest way to install, compile, and run our code is through the use of Docker.\
+If you want to exploit the real-time capabilities of our method however, we recommend that you compile and install the code directly on your machine (see the "without Docker" section below).
+
+Simply use the given Dockerfile to install all the dependencies and compile the code:
+
+```txt
+docker build -t rt_of_low_high_res_event_cameras .
+```
+
+Then, to run the built container interactively, we recommend the use of [rocker](https://github.com/osrf/rocker), which can be installed on the host with pip:
+
+```txt
+pip install rocker
+```
+
+Once done, the following command can be used to launch the container with GPU support, X11 forwarding, and access to files in the home folder:
+
+```txt
+rocker --nvidia --x11 --home rt_of_low_high_res_event_cameras
+```
+
+## Installing and compiling the code (without Docker)
 
 This code has originally been developed and tested with Ubuntu 20.04, ROS Noetic, CUDA 11.5+ and OpenCV 4.2.\
 It is also compatible with Ubuntu 18.04, ROS Melodic, CUDA 11.5+ and OpenCV 3.2.\

